qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -0,0 +1,352 @@
+            inactive_orbs_idxs = list(range(norbs_inactive))
+            active_orbs_idxs = list(range(norbs_inactive, norbs_inactive+self.num_orbitals))
+        else:
+            active_orbs_idxs = self.active_orbitals
+            inactive_orbs_idxs = [o for o in range(nelec_total // 2) if o not in
+                                  self.active_orbitals and mo_occ_total[o] > 0]
+
+        # split molecular orbitals coefficients into active and inactive parts
+        mo_coeff_inactive = (mo_coeff_full[0][:, inactive_orbs_idxs],
+                             mo_coeff_full[1][:, inactive_orbs_idxs] if beta else None)
+        mo_coeff_active = (mo_coeff_full[0][:, active_orbs_idxs],
+                           mo_coeff_full[1][:, active_orbs_idxs] if beta else None)
+        mo_occ_inactive = (mo_occ_full[0][inactive_orbs_idxs],
+                           mo_occ_full[1][inactive_orbs_idxs] if beta else None)
+
+        density_inactive = self._compute_inactive_density_matrix(mo_occ_inactive, mo_coeff_inactive)
+
+        # extract core Hamiltonian and electron-repulsion-integral matrices from QMolecule
+        hcore = (q_molecule.hcore, q_molecule.hcore_b if beta else None)
+        eri = q_molecule.eri
+
+        fock_inactive = self._compute_inactive_fock_op(hcore, eri, density_inactive)
+
+        e_inactive = self._compute_inactive_energy(hcore, density_inactive, fock_inactive,
+                                                   mo_coeff_inactive)
+
+        hij, hijkl = self._compute_active_integrals(mo_coeff_active, fock_inactive, eri)
+
+        # construct new QMolecule
+        q_molecule_reduced = copy.deepcopy(q_molecule)
+        # Energies and orbits
+        q_molecule_reduced.energy_shift['inactive_energy'] = e_inactive
+        q_molecule_reduced.num_orbitals = self.num_orbitals
+        q_molecule_reduced.num_alpha = num_alpha
+        q_molecule_reduced.num_beta = num_beta
+        q_molecule_reduced.mo_coeff = mo_coeff_active[0]
+        q_molecule_reduced.mo_coeff_b = mo_coeff_active[1]
+        q_molecule_reduced.orbital_energies = q_molecule.orbital_energies[active_orbs_idxs]
+        if beta:
+            q_molecule_reduced.orbital_energies_b = q_molecule.orbital_energies_b[active_orbs_idxs]
+        # 1 and 2 electron integrals in MO basis
+        q_molecule_reduced.mo_onee_ints = hij[0]
+        q_molecule_reduced.mo_onee_ints_b = hij[1]
+        q_molecule_reduced.mo_eri_ints = hijkl[0]
+        q_molecule_reduced.mo_eri_ints_ba = hijkl[1]
+        q_molecule_reduced.mo_eri_ints_bb = hijkl[2]
+        # invalidate AO basis integrals
+        q_molecule_reduced.hcore = None
+        q_molecule_reduced.hcore_b = None
+        q_molecule_reduced.kinetic = None
+        q_molecule_reduced.overlap = None
+        q_molecule_reduced.eri = None
+        # invalidate dipole integrals
+        q_molecule_reduced.x_dip_ints = None
+        q_molecule_reduced.y_dip_ints = None
+        q_molecule_reduced.z_dip_ints = None
```

Comment:
So this method does not allow the dipole ints to be transformed accordingly meaning we cannot measure dipole when using active space reduction. In which case maybe we want to have the approximate reduction too since that allowed it.

Reply:
I agree. But I will introduce another transformer implementation which does this in a separate PR.